### ISSUE TITLE
predictions to label?

### ISSUE THREAD

**Issue by preke** [lsrock1/abcnn_pytorch#2]
Forgive me for another question...
If I was trying to use the model for text classification.
How I transfer the predictions to label?
I mean, the predictions are a bunch of float numbers.
If the labels are 0s and 1s(although they can be float numbers, too)
As the prediction are sigmoided, am I supposed to use 0.5 as the threshold?

**Comment by lsrock1** [comment]
I have a lot of times, welcome your question.
I have used this model to sentence similarity prediction, (1: similar, 0: different)
In my experiment
1. when training, input -> model -> Binary Cross Entropy loss with logit(in pytorch)
loss is computed by considering float output, therefore you dont need to think threshold.
with logit means including sigmoid, so you dont have to include sigmoid in model(They said BCELoss with logit is more numerical stable. It is better to use BCELosswithlogit than Sigmoid and BCELoss seperately)

2. when inference,
yes, like you said, decide the threshold and make your float outputs integer(0 or 1)

**Comment by preke** [comment]
Yes, I also read them in the docs of pytorch.
And as you said, I decide to choose a threshold to change the output to integer labels.
It is really kind of you to answer me detailedly.
Thank you!
